### PR TITLE
feat: send once option for threshold alerts

### PR DESCRIPTION
**Pull request by magnew**
Closes: #9026 

### Description:

Note that this is a draft at the moment. It seems to work, but I'd like to test it more rigorously. 

Adds a checkbox to send scheduled alerts just once. This does this by disabling the alert once it's been sent. I added some helper text when the box is turned off. Should we add some text when it is on as well.

Send once state:
Screenshot 2024-02-23 at 17 57 43

Send always state:
Screenshot 2024-02-23 at 17 57 59

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by magnew** [comment]
> * I think I prefer to see this with the Alert conditions section? So perhaps at the bottom of it and spaced so it feels like part of that section. Do you also think it makes sense there?

Sure, let's give that a shot

File: packages/backend/src/scheduler/SchedulerTask.ts [hunk 1]
```diff
@@ -1452,7 +1454,22 @@ export const handleScheduledDelivery = async (
                     savedChartUuid,
                 );
 
-                if (isPositiveThesholdAlert(thresholds, rows)) {
+                if (isPositiveThresholdAlert(thresholds, rows)) {
+                    // If the delivery frequency is once, we disable the scheduler.
+                    // It will get sent once this time.
+                    if (
+                        notificationFrequency === NotificationFrequency.ONCE &&
```
Comment (rephus) [inline packages/backend/src/scheduler/SchedulerTask.ts:1461]: Nice

Comment (rephus) [reply packages/backend/src/scheduler/SchedulerTask.ts:1461]: this means some scheduler events can happen even when `scheduler` is disabled, like send email/slack notifications. We do the disabled check here https://github.com/lightdash/lightdash/pull/9089/files#diff-76827588cb3d2be92b5d1fec14b04f04caf22bbcd510e49a98517da3192d3fcbR1416 so it should not cause any issues. Just something to keep in mind.

**Review by rephus — COMMENTED** [review]
Code looks good, waiting for render to be fixed to test it

**Comment by magnew** [comment]
> Could you fix the migration issue in render ? this happens if the `migration` file was not in the first commit...

Sure, I kicked up a database reset/migration. That should do it right?

File: packages/backend/src/scheduler/SchedulerTask.ts (hunk 1 above)
Comment (magnew) [reply packages/backend/src/scheduler/SchedulerTask.ts:1461]: Good point. I agree I *think* it should be safe. I could make this just set a local variable like disableAfterSending and disable it at the end of this try block. Would that be better or more confusing for DevX?

**Review by rephus — APPROVED** [review]
Yeah, it works as expected. 

It is true `enabling` the toggle on render is very slow :thinking: we should fix it somehow... 

This was not introduced by this PR and it not a blocker anyway.

**Pull request merged**